test/unit/models/transaction.test.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * This eslint rule is disabled because of https://github.com/nodesecurity/eslint-plugin-security/issues/21
+ * It gives linting errors in testing the DB data with keys from fixtures
+ */
+/* eslint-disable security/detect-object-injection */
+
+const chai = require('chai')
+const { expect } = chai
+
+const cleanDb = require('../../utils/cleanDb')
+const transactionModelsImpl = require('../../../models/transaction')
+const transactionDataArray = require('../../fixtures/transaction/transactions')()
+const firestore = require('../../../utils/firestore')
+const transactionsModel = firestore.collection('transaction')
+
+/**
+ * Test the model functions and validate the data stored
+ */
+
+describe('transaction', function () {
+  before(function () {
+    transactionDataArray.forEach(async (element, index) => {
+      await transactionsModel.add(transactionDataArray[index])
+    })
+  })
+  after(async function () {
+    await cleanDb()
+  })
+
+  describe('when fetchTransactionsByUserId called, it ', function () {
+    it('in DESC order', async function () {
+      const expectedTransactionData = transactionDataArray.sort((a, b) => {
+        return new Date(b.dateTime) - new Date(a.dateTime)
```

Comment:
The timestamp could be directly sorted, why convert it into data time than sort it? any specific reasons?

Reply:
No reasons as such. They are now removed.